File: API/UniUnboxdAPI/Models/DataTransferObjects/RegisterModel.cs
using System.Text.Json.Serialization;

namespace UniUnboxdAPI.Models.DataTransferObjects
{
    // RegisterModel class represents the data transfer object for user registration.
    public class RegisterModel
    {
        // Email property represents the email address of the user.
        public required string Email { get; set; }

        // Password property represents the password of the user.
        public required string Password { get; set; }

        // Type property represents the type of user (e.g., student, professor).
        [JsonConverter(typeof(JsonStringEnumConverter))] 
        public required UserType Type { get; set; }
    }
}

File: Application/app/src/main/java/com/example/uniunboxd/DTO/RegisterModel.java
package com.example.uniunboxd.DTO;

/**
 * The RegisterModel class represents the data required for user registration.
 * It includes an email, a password, and a user type.
 */
public class RegisterModel {
    /**
     * The email of the user.
     */
    public String email;

    /**
     * The password of the user.
     */
    public String password;

    /**
     * The type of the user (e.g., "student", "professor").
     */
    public String userType;

    /**
     * Constructs a new RegisterModel with the specified email, password, and user type.
     * @param email The email of the user.
     * @param password The password of the user.
     * @param userType The type of the user.
     */
    public RegisterModel(String email, String password, String userType) {
        this.email = email;
        this.password = password;
        this.userType = userType;
    }
}
